functions/auth0.py

```diff
@@ -88,12 +89,36 @@ def _get_log_bucket():
 
 def _get_last_auth0_log_id() -> Optional[str]:
     """Fetch that ID of the last access log imported from Auth0"""
-    blob = _get_log_bucket().get_blob(LAST_LOG_ID)
-    if blob:
-        return blob.download_as_string().decode("utf-8")
+    bucket = _get_log_bucket()
+    blobs = bucket.list_blobs(prefix=LAST_LOG_ID_DIR)
+
+    last_blob = None
+
+    for blob in blobs:
+        if not last_blob:
+            last_blob = blob
+            continue
+
+        if blob.name > last_blob.name:
+            last_blob = blob
```

Comment:
Pretty sure this code block is equivalent to 
```python
last_blob = max(blobs, key=lambda b: b.name) if blobs else None
```
Also, can you add a comment about why comparing the names in this way guarantees we get the last log ID?